### ISSUE TITLE
Unable to build wbox toolchain w/ GCC 10

### ISSUE THREAD

**Issue by YoshiRulz** [TASEmulators/BizHawk#2865]
At 09afbdd6e (submodule at nattthebear/musl@76d5e7a4b). Going through the steps in the readme, I can't progress past building musl. I have GCC 10.2.0 on Manjaro.
Shell history:
```sh
git submodule update --init llvm-project/
git submodule update --init musl/
cd waterbox/musl/
./wbox_configure.sh 
./wbox_build.sh
```
Tail of output of `./wbox_build.sh`:
```
[...]
gcc -std=c99 -nostdinc -ffreestanding -fexcess-precision=standard -frounding-math -Wa,--noexecstack -D_XOPEN_SOURCE=700 -I./arch/waterbox -I./arch/generic -Iobj/src/internal -I./src/include -I./src/internal -Iobj/include -I./include  -Os -pipe -fomit-frame-pointer -fno-unwind-tables -fno-asynchronous-unwind-tables -ffunction-sections -fdata-sections -Werror=implicit-function-declaration -Werror=implicit-int -Werror=pointer-sign -Werror=pointer-arith -Werror -fvisibility=hidden -mcmodel=large -mstack-protector-guard=global -no-pie -fno-pic -fno-pie -c -o obj/src/unistd/getcwd.o src/unistd/getcwd.c
src/unistd/getcwd.c: In function ‘getcwd’:
cc1: error: function may return address of local variable [-Werror=return-local-addr]
src/unistd/getcwd.c:9:7: note: declared here
    9 |  char tmp[buf ? 1 : PATH_MAX];
      |       ^~~
cc1: all warnings being treated as errors
make: *** [Makefile:149: obj/src/unistd/getcwd.o] Error 1
```

**Comment by Morilli** [comment]
I would recommend just silencing this warning in that spot for now. It is obviously an incorrect warning which stems from a bug in gcc 10+. See [this bug report](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=93644) at gcc which I believe is this very issue.

Might be worth to make an upstream musl issue also, idk.

**Comment by Morilli** [comment]
Resolved in ddac50eb3014c69b31dd9933ff7170b05dd207c0.
